### DISCUSSION TITLE
Opengrok notices newer files but does not rebuild the xref .gz files

### DISCUSSION THREAD

**Discussion by dkr91** [oracle/opengrok#4626] [Q&A]
**Describe the bug**
Opengrok notices newer files but does not rebuild the xref .gz files

If you provide the exact version of OpenGrok, JDK used, OS(and its version) used and Tomcat(or your webapp server) used it can help figuring out an environment issue.
Opengrok Docker 1.13.7

**To Reproduce**
Let `opengrok-sync` update the source.

**Expected behavior**
Opengrok checkFileInner updates the `.gz` files when it notces a updated code-file

**Screenshots**
```log
14-May-2024 09:07:13.608 INFO [ForkJoinPool.commonPool-worker-945] org.opengrok.suggest.Suggester.rebuild Suggesters for [demo] were rebuilt (took 4 minutes 57 seconds)
14-May-2024 09:07:26.367 WARNING [http-nio-8080-exec-2] org.opengrok.web.PageConfig.checkFileInner file '/opengrok/src/demo/src/test.sh' is newer than '/opengrok/data/xref/demo/src/test.sh.gz'
```
**Additional context**

**Comment by vladak** [comment]
These is an excerpt from the webapp log. What might have happened is that the request to the webapp for the file arrived just after the file was updated on disk and before the indexer ran. You can raise log level for the indexer to see how it handled the file and when. There is a way how to make reindexing almost atomic however it requires sort of elaborate workflow and possibly a file-system capable of creating snapshots/clones.

Otherwise it would be a bug however much more information would be needed in such case.
